app/javascript/lib/tagging/index.js

```diff
@@ -57,30 +59,53 @@ export function initMultiSelectVitaPartner() {
                     role="option">
                         <div class='${item.parentName ? "parent" : ""}'>${html`${item.parentName || ''}`}</div>
                         <div class='${item.parentName ? "site" : "org"}'>${html`${item.value}`}</div>
-                    </div>`
+                    </div>`;
             },
         }
     });
 
     input.addEventListener("change", (event) => {
-        let { value } = event.target
+        // extracts just the vita partner ids and stores in hidden field
+        let { value } = event.target;
         if (value) {
-            value = JSON.stringify(JSON.parse(event.target.value).map(v => v.id))
+            value = JSON.stringify(JSON.parse(event.target.value).map(v => v.id));
+        }
+
+        const hiddenField = document.querySelector("#vita_partners");
```

Comment:
I tried to delete this but then ran into all sorts of issues because some of the forms that use `multi-select-vita-partner` have a hidden field (something that might look like `<%= f.hidden_field(:vita_partners, class: "multi-select-vita-partner") %>`) where they will store these vita partner ids for the form like on admin/experiments or client filters

Reply:
i think this refactor makes things clearer, ty!